### PR TITLE
Add redshift maintenance operation

### PR DESCRIPTION
**Pull request by clrcrl**
My flight from Dublin got delayed. Decided to beat jet lag by working on this.

### CODE REVIEW

File: macros/redshift_maintenance_operation.sql [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{% macro get_vacuumable_tables() %}
```
Comment (clrcrl) [inline macros/redshift_maintenance_operation.sql]: We could parameterize this further, say, have a schema condition, or a max column width condition.

I am not too worried about doing this in the package though -- anyone who is doing this sort of work should feel comfortable about how to add a macro to their own project.\

File: macros/redshift_maintenance_operation.sql [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    {% for table in get_vacuumable_tables() %}
+        {% set start=modules.datetime.datetime.now() %}
+        {% set message_prefix=loop.index ~ " of " ~ loop.length %}
+        {{ dbt_utils.log_info(message_prefix ~ " Vacuuming " ~ table) }}
+        {% set results=run_query("vacuum " ~ table) %}
+        {{ dbt_utils.log_info(message_prefix ~ " Analyzing " ~ table) }}
+        {% do run_query("analyze " ~ table) %}
+        {% set end=modules.datetime.datetime.now() %}
+        {% set total_seconds = (end - start).total_seconds() | round(2)  %}
```
Comment (clrcrl) [inline macros/redshift_maintenance_operation.sql]: this is totally extra, i'm so sorry lol

File: macros/redshift_maintenance_operation.sql [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{% macro get_vacuumable_tables() %}
+    {% set vacuumable_tables_sql %}
+        select
+            '"' || table_schema || '"."' || table_name || '"' as table_name
+        from information_schema.tables
```
Comment (drewbanin) [inline macros/redshift_maintenance_operation.sql]: i imagine we don't want to vacuum every relation returned here -- we should probably exclude:
1. views (only where `table_type == 'BASE TABLE'`)
2. everything in the information schema (`table_schema in ('information_schema', 'pg_catalog')`)

File: macros/redshift_maintenance_operation.sql (hunk 2 above)
Comment (drewbanin) [reply macros/redshift_maintenance_operation.sql]: i support it :)

**Pull request merged**